src/lib.rs

```diff
@@ -115,118 +117,165 @@ pub fn run() -> Result<()> {
     {
         let mut state = STATE.lock().unwrap();
 
-        state.input_mode = config.input_mode;
-        state.dial_rows = dial_rows;
-        state.ball = Ball::new(
-            config.ball.random_direction_change_time_min,
-            config.ball.random_direction_change_time_max,
-            config.ball.velocity_meter,
-        );
-        state.session_output = SessionOutput::new(
-            config
-                .output_data_path
-                .clone()
-                .unwrap_or_else(|| String::from(DEFAULT_OUTPUT_PATH)),
-        );
+        if let AppState::Running(state) = state.deref_mut() {
+            // Assign all of the values that we have created from the configuration file
+            // because these had to come with defaults since it is static
+            state.input_mode = config.input_mode;
+            state.dial_rows = dial_rows;
+            state.ball = Ball::new(
+                config.ball.random_direction_change_time_min,
+                config.ball.random_direction_change_time_max,
+                config.ball.velocity_meter,
+            );
+            state.session_output = SessionOutput::new(
+                config
+                    .output_data_path
+                    .clone()
+                    .unwrap_or_else(|| String::from(DEFAULT_OUTPUT_PATH)),
+            );
+        } else {
+            panic!();
```

Comment:
nit: use `unreachable!("App always in the running state on startup")`?

Reply:
unreachable gets completely taken out in release builds, which I'm not sure if I'm comfortable with the program continuing in any case, even if we test it in debug mode. I'll change it to panic with a message though.